action/builder.go

```diff
@@ -84,6 +84,11 @@ type EnvelopeBuilder struct {
 	ab  AbstractAction
 }
 
+func (b *EnvelopeBuilder) SetTxType(v uint32) *EnvelopeBuilder {
+	b.ab.txType = v
+	return b
+}
```

Comment:
as chenchen suggested in previous pr, action builders by types may be a better way.

Reply:
can do in another PR since changes will be quite a lot